src/mpid/ch4/netmod/ofi/ofi_events.h

```diff
@@ -405,15 +405,17 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_get_huge_event(struct fi_cq_tagged_entry
 
         if (bytesToGet == 0ULL) {
             MPIDI_OFI_send_control_t ctrl;
+            /* recv->localreq may be freed during done_fn.
+             * Need to backup the handle here for later use with MPIDI_CH4U_map_erase. */
+            uint64_t key_to_erase = recv->localreq->handle;
```

Comment:
Is there a limitation to calling `map_erase` here instead?

Reply:
It should probably work too. I was lazy to make sure nothing between here and L418 refers to the map. Would prefer having it here?

Reply:
After looking again, I prefer your version since it's more consistent to free everything at the end.